tests/test_makeLimitedFitsHeader.py

```diff
@@ -63,17 +91,43 @@ def testBasics(self):
         header = makeLimitedFitsHeader(metadata)
 
         expectedLines = [  # without padding to 80 chars
-            "ABOOL   = 1",
+            "ABOOL   = T",
             "AFLOAT  =              1.2E+25",
+            "AFLOAT2 =                1E+30",
             "ANINT   =                   -5",
+            "AFLOATZ =                    0.0",
+            "INTFLOAT=                   -5.0",
+            "LONGFLT = 0.0089626337538440005",
             "ASTRING1= 'value for string'",
         ]
         expectedHeader = "".join("%-80s" % val for val in expectedLines)
 
-        self.assertEqual(header, expectedHeader)
+        self.assertHeadersEqual(header, expectedHeader, rtol=1e-16)
 
         self.checkExcludeNames(metadata, expectedLines)
 
+    def testSinglePrecision(self):
+        """Check that single precision floats do work"""
+        metadata = PropertyList()
+
+        # Numeric form of single precision floats need smaller precision
+        metadata.setFloat("SINGLE", 3.14159)
+        metadata.setFloat("SINGLEI", 5.0)
+        metadata.setFloat("SINGLEE", -5.9e20)
+        metadata.setFloat("EXP", -5e10)
+
+        header = makeLimitedFitsHeader(metadata)
+
+        expectedLines = [  # without padding to 80 chars
+            "SINGLE  =              3.14159",
+            "SINGLEI =                  5.0",
+            "SINGLEE =             -5.9E+20",
+            "EXP     =               -5E+10",
+        ]
+        expectedHeader = "".join("%-80s" % val for val in expectedLines)
+
+        self.assertHeadersEqual(header, expectedHeader, rtol=1e-7)
```

Comment:
consider an rtol based on floating point information e.g. `rtol=np.finfo(np.float32).resolution` here and for double precision -- just to make it a bit clearer why you chose these values of rtol. A brief code comment would probably suffice, but finfo makes it truly obvious.